more_itertools/more.py

```diff
@@ -3313,8 +3311,8 @@ def set_partitions(iterable, k=None, min_size=None, max_size=None):
     ['a', 'b', 'c']
 
     """
-    L = list(iterable)
-    n = len(L)
+    lst = list(iterable)
```

Comment:
This is another dubious ruff rule.  `L` and `M` are perfectly valid variable names.